project-addons/purchase_picking/models/stock.py

```diff
@@ -207,17 +196,17 @@ def reassign_reservation_dates(self, product_id):
             [('product_id', '=', product_id.id),
              ('state', '=', 'draft'),
              ('picking_type_id.code', '=', u'incoming')],
-             order='date_expected')
+            order='date_expected')
         reservation_index = 0
 
         reservation_used = 0
         for move in moves:
             qty_used = 0
-            product_uom_qty = uom_obj._compute_qty_obj(move.product_uom, move.product_uom_qty, product_uom)
+            product_uom_qty = uom_obj._compute_quantity(move.product_uom_qty, product_uom)
```

Comment:
Esto creo que no estaría bien, no se puede ejecutar contra la clase habría que ejecutarlo contra una instancia, creo que sería: move.product_uom._compute_quantity(move.product_uom_qty, product_uom), convierte la cantidad de una unidad de medida a otra, creo que no os afectaría en vuestro caso porque sólo usáis unidad, como unidad de medida y no hacéis conversiones pero bueno sería mejor dejarlo corregido por si en el futuro se necesita. Abajo se hace lo mismo.